// @okta/okta-angular/esm2015/src/okta/has-any-group.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { OktaAuthStateService } from './services/auth-state.service';
let OktaHasAnyGroupDirective = class OktaHasAnyGroupDirective {
    constructor(
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    templateRef, viewContainer, authStateService) {
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
        this.authStateService = authStateService;
    }
    set oktaHasAnyGroup(groups) {
        this.authStateService.hasAnyGroups(groups)
            .subscribe(isAuthorized => {
            // don't update UI if no state change
            if (isAuthorized === this.previousIsAuthorized) {
                return;
            }
            this.previousIsAuthorized = isAuthorized;
            this.viewContainer.clear();
            if (isAuthorized) {
                this.viewContainer.createEmbeddedView(this.templateRef);
            }
        });
    }
};
__decorate([
    Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], OktaHasAnyGroupDirective.prototype, "oktaHasAnyGroup", null);
OktaHasAnyGroupDirective = __decorate([
    Directive({ selector: '[oktaHasAnyGroup]' }),
    __metadata("design:paramtypes", [TemplateRef,
        ViewContainerRef,
        OktaAuthStateService])
], OktaHasAnyGroupDirective);
export { OktaHasAnyGroupDirective };
